### PR TITLE
[AnyAddress] Switch by blockchain type first

### PR DESCRIPTION
**Pull request by hewigovens**
Fixes https://github.com/trustwallet/wallet-core/issues/2281, easier to add new cosmos sdk / evm chain

### CODE REVIEW

File: src/AnyAddress.h [hunk 1]
```diff
@@ -17,15 +18,16 @@
...
 #include "Elrond/Address.h"
+#include "Ethereum/Address.h"
 #include "Kusama/Address.h"
 #include "NEAR/Address.h"
 #include "NEO/Address.h"
 #include "Nano/Address.h"
 #include "Polkadot/Address.h"
 #include "Ronin/Address.h"
 #include "Solana/Address.h"
+#include "TrustWalletCore/TWBlockchain.h"
```
Comment (catenocrypt) [inline src/AnyAddress.h:30]:
```suggestion
#include <TrustWalletCore/TWBlockchain.h>
```

**Review by catenocrypt — APPROVED** [review]

File: src/AnyAddress.h [hunk 2]
```diff
@@ -34,125 +36,96 @@
...
-            auto addr = Bitcoin::BitcoinCashAddress(string);
-            data.resize(Bitcoin::Address::size);
-            size_t outlen = 0;
-            cash_data_to_addr(data.data(), &outlen, addr.bytes.data(), 34);
-            return {data.begin() + 1, data.end()};
-        }
+            case TWCoinTypeBitcoinCash: {
+                Data data;
+                auto addr = Bitcoin::BitcoinCashAddress(string);
+                data.resize(Bitcoin::Address::size);
```
Comment (Milerius) [inline src/AnyAddress.h:73]: Do you really want to use `resize` here? `resize` appends the default element to the vector, and `reserve` grow the capacity but let the data uninitialized.

If you really want to use `resize` then you can write `Data data(Bitcoin::Address::size)` will allow to use the constructor with pre-defined size from `std::vector` 

from https://en.cppreference.com/w/cpp/container/vector/vector

File: src/AnyAddress.h [hunk 3]
```diff
@@ -34,125 +36,96 @@
...
-            auto addr = Bitcoin::ECashAddress(string);
-            data.resize(Bitcoin::Address::size);
-            size_t outlen = 0;
-            cash_data_to_addr(data.data(), &outlen, addr.bytes.data(), 34);
-            return {data.begin() + 1, data.end()};
-        }
+            case TWCoinTypeECash: {
+                Data data;
+                auto addr = Bitcoin::ECashAddress(string);
+                data.resize(Bitcoin::Address::size);
```
Comment (Milerius) [inline src/AnyAddress.h:82]: Same comment as above

**Comment by catenocrypt** [comment]
See also #2269

File: src/AnyAddress.h (hunk 1 above)
Comment (hewigovens) [reply src/AnyAddress.h:30]: This is added automatically, let me fix it

File: src/AnyAddress.h (hunk 2 above)
Comment (hewigovens) [reply src/AnyAddress.h:73]: This is legacy code, good suggestion

**Comment by hewigovens** [comment]
Looks single Abi works best on CI

**Pull request merged**